### PR TITLE
Pract.02: Add 01-03 | Nikita Beliy

### PR DESCRIPTION
**Pull request by N1k1ta-white**
Nikita Beliy
SI 1 9MI800022

### CODE REVIEW

File: Task01.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+#include <fstream>
+
+const char PATH[] = "result.txt";
+
+int diff_betw_sum_comp() {
+    std::fstream my_file(PATH, std::ios::in);
+    int sum, comp;
```
Comment (khatoni) [inline Task01.cpp:10]: Check whether the file has opened successfully

File: Task02.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#include <iostream>
+#include <fstream>
+#include "assert.h"
+
+int lengthOfFile(const char path[]) {
+    std::fstream read_file(path, std::ios::in);
+    assert(read_file.is_open());
+    int length = 0;
+    char rotator;
+    while(read_file >> std::noskipws >> rotator)
```
Comment (khatoni) [inline Task02.cpp]: You are not allowed to use std::noskipws

File: Task03.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    char sender_name[MAX_NAME_SIZE];
+};
+
+Info get_info_from_file(const char path[]) {
+    std::fstream file(path, std::ios::in);
+    assert(file.is_open());
+    Info res{};
+    char rotator;
+    int line = 0;
+    while (file >> std::noskipws >> rotator) {
```
Comment (khatoni) [inline Task03.cpp]: What is std::noskipws?

File: Task03.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        if (rotator == ' ') {
+            switch (line) {
```
Comment (khatoni) [inline Task03.cpp]: Read again the task, you can not be sure that they will be in this order!

File: Task03.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                    break;
+                case 2:
+                    new_file << info.message;
+                    break;
+                case 3:
+                    new_file << info.sender_name;
+                    break;
+                default:
+                    break;
+            }
```
Comment (khatoni) [inline Task03.cpp]: Same as in the function above

**Review by khatoni — CHANGES_REQUESTED** [review]

**Comment by Justsvetoslavov** [comment]
Check uploaded solutions.
You can re-open the PR or open a new one if you have better solution or questions.

**Pull request closed without merging**